inference-engine/src/transformations/src/transformations/common_optimizations/remove_loop_dangling_parameters.cpp

```diff
@@ -0,0 +1,63 @@
+// Copyright (C) 2018-2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "transformations/common_optimizations/remove_loop_dangling_parameters.hpp"
+#include "transformations/utils/utils.hpp"
+
+#include <memory>
+#include <vector>
+
+#include <ngraph/opsets/opset8.hpp>
+#include <ngraph/op/util/multi_subgraph_base.hpp>
+#include <ngraph/rt_info.hpp>
+#include <ngraph/pattern/op/wrap_type.hpp>
+#include "itt.hpp"
+
+
+NGRAPH_RTTI_DEFINITION(ngraph::pass::RemoveLoopDanglingParameters, "RemoveLoopDanglingParameters", 0);
+
+ngraph::pass::RemoveLoopDanglingParameters::RemoveLoopDanglingParameters() {
```

Comment:
Added support for all `MultiSubgraphOp`.
@itikhono @jane-intel General question: supporting "If" case makes code even more complicated than before, especially because If's input can be used also in `then` and `else` body so additional logic is needed.
Is ok for you? (or maybe do you have some idea for simpler implementation?)

Reply:
@mbencer in general, looks good
I have no idea how to simplify this.